src/lib/util.js

```diff
@@ -351,37 +375,21 @@ export function appendAuthParams(url, token = '', sharedLink = '', password = ''
         return url;
     }
 
-    let delim = '?';
+    const uri = new Uri(url);
 
-    if (url.indexOf('?') > 0) {
-        delim = '&';
-    }
-
-    let params = '';
     if (token) {
-        params = `access_token=${token}`;
+        uri.addQueryParam('access_token', token);
     }
 
     if (sharedLink) {
-        if (params) {
-            params = `${params}&`;
-        }
-        params = `${params}shared_link=${encodeURI(sharedLink)}`;
+        uri.addQueryParam('shared_link', sharedLink);
+
         if (password) {
-            params = `${params}&shared_link_password=${encodeURI(password)}`;
+            uri.addQueryParam('shared_link_password', password);
         }
     }
 
-    // Following params are for API analytics
-    if (CLIENT_NAME) {
-        params = `${params}&${PARAM_CLIENT_NAME}=${encodeURI(CLIENT_NAME)}`;
-    }
-
-    if (CLIENT_VERSION) {
-        params = `${params}&${PARAM_CLIENT_VERSION}=${encodeURI(CLIENT_VERSION)}`;
-    }
-
-    return `${url}${delim}${params}`;
+    return appendQueryParams(uri.toString(), ANALYTICS_QUERY_PARAMS);
```

Comment:
A little confusing that addQueryParam takes a URI while appendQueryParam takes a string. Not much we could do since add is coming from the library. Maybe create a small wrapper in util to make things consistent?